mindsdb/api/mysql/mysql_proxy/classes/sql_query.py

```diff
@@ -1161,6 +1161,16 @@ def execute_step(self, step, steps_data):
                 table_name = get_preditor_alias(step, self.database)
                 columns_collection = ColumnsCollection()
                 dn = self.datahub.get(self.mindsdb_database_name)
+                base_ml_engine_type_dict = {}
+                if ml_handler_name not in ("mindsdb", "lightwood"):
+                    base_ml_engine_type_dict = dn.get_base_ml_engine_table_column_dict(table_name=table_name,
+                                                                   ml_handler_name=ml_handler_name)
+                    if base_ml_engine_type_dict is not None:
+                        for col, col_type in base_ml_engine_type_dict.items():
+                            columns_collection.add(table_name, (col, col))
+                    else:
+                        base_ml_engine_type_dict = {}
+
```

Comment:
Do I understand right: it was done to add columns to result from predictor?  But you can just return this columns from predict method and they will be added to result

Reply:
If I only return these columns in prediction method but without the modification in sql_query.py, exception will be raised in the phase of sql-parsing.

Reply:
What exception do you have in sql_query.py without this block?

I made a little test: removed all your changes from sql_query, added coumn at the end of predict method in handler.
I see no error in sql_query, this column is passed to result and test are also working  
![image](https://user-images.githubusercontent.com/8502631/198241565-5763be19-1efe-4ecf-98d0-f8b33ff93e08.png)
![image](https://user-images.githubusercontent.com/8502631/198241619-73b281a9-6ac2-4151-8c55-01bd03e69fdd.png)

Reply:
hey bros, I nearly can't believe my eyeball. I deleted all my adding code in sql_query and other related codes, run the end-to-end test, passed! run the test cases, passed, Again!!!

My dream of being super programmer ended ... however, I am very clearly remember that it would crash without that codes.

Anyway,  forget that, I've refined the codes, please have a check. Forgive me ... Thank you!